src/app/charts/line-graph.component.ts

```diff
@@ -97,12 +106,12 @@ export class LineGraphComponent {
     // Set axis scales
     private setAxisScales(): void {
         // Time scales only recognize annual and daily data
-        var parseTime = D3.timeParse('%Y'); //D3.timeParse('%Y-%m-%d');
+        var parseTime = D3.timeParse(this.timeFormat);
         this.extractedData.forEach(d => d.date = parseTime(d.date));
         this.xRange = D3.extent(this.extractedData, d => d.date);
         this.xScale.domain(this.xRange);
         // Adjust y scale, prettify graph
-        let yPad = Math.abs((D3.max(this.yData) - D3.min(this.yData)) * 2/3);
+        const yPad = ((D3.max(this.yData) - D3.min(this.yData)) > 0) ? ((D3.max(this.yData) - D3.min(this.yData)) * 2/3) : 5;
```

Comment:
Actually meant to suggest 2/3 be made a constant, but I suppose this works too. Generally numbers in code should be assigned to constants for readability. (What does 2/3 do here? Is that a spacing amount?)